pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java

```diff
@@ -68,16 +70,22 @@ public void expireMessages(int messageTTLInSeconds) {
                     messageTTLInSeconds);
 
             cursor.asyncFindNewestMatching(ManagedCursor.FindPositionConstraint.SearchActiveEntries, entry -> {
-                MessageImpl<?> msg = null;
+                Pair<MessageImpl<byte[]>, PulsarApi.BrokerEntryMetadata> pair = null;
                 try {
-                    msg = MessageImpl.deserialize(entry.getDataBuffer());
-                    return msg.isExpired(messageTTLInSeconds);
+                    pair = MessageImpl.deserializeWithBrokerEntryMetaData(entry.getDataBuffer());
```

Comment:
We should expose the broker metadata in the Message. So this would avoid using `Pair` and a lot of `if-else` logic.

We can improve `msg.isExpired` logic. If entry metadata is present, use broker timestamp; otherwise use client timestamp.

Reply:
Nice susggestion.  I'll optimize the implement : 
1. expose entry metadata to MessageImpl to avoid `if-else` logic
2. skip deserialize the MessageMetadata if Entry has entry metadata

Reply:
Nice suggestion.  I'll optimize the implement as follow steps:
1. expose the entry metadata to MessageImpl
2. if entry metadata exist, skip the deserialization of MessageMetadata
3. if not exist, deserialization the MessageMetadata